lib/classes/Variables.test.js

```diff
@@ -1892,15 +1903,15 @@ module.exports = {
         .should.become({ hello: 'hello world' });
     });
 
-    it('should throw if property exported by a javascript file is not a function', () => {
+    it('should populate an entire variable exported by a javascript file object', () => {
```

Comment:
This seems a duplicate of test above at 1884 now.  Are there no error conditions?  You seem to assume, given your language in the code, that only objects would be returned.  Are there no cases where concrete values will be returned?  What about promises?  This test should probably stay concentrated on an expected failure and perhaps be broken up in to multiple sad path oriented paths.

Reply:
> This seems a duplicate of test above at 1884 now.

It's definitely a similar test case to 1884, though I was seeing this as sort of a sibling of the 1895 case.  1873 and 1884 are both referencing the `hello` attribute of the exported file (`file(./hello.js):hello`) while 1895 and 1906 take the file wholesale (`file(./hello.js)`).  If you feel strongly that this is still a duplication of test cases, I'm happy to remove it.

> Are there no error conditions?

Likely the answer is "yes, there are", though it appears there aren't a lot of test cases currently probing at error handling when it comes to reading from a file.  I can take a look at what sorts of potential errors we may be introducing here, but if there are any that come to mind, please let me know.  Perhaps it's as simple as making sure we propagate any errors appropriately.

> You seem to assume, given your language in the code, that only objects would be returned. Are there no cases where concrete values will be returned?

Don't see why not.  I'll add a test case for this.

> What about promises?

It looks like functions that return a promise are already being covered by tests around line 1034 (though maybe I'm misunderstanding the purpose of these test).  If we mean just simply exporting a promise, I can look into what effect that has, though I don't think this 
 would be a new issue - a function could return an object that contains promises within it.

Reply:
> If you feel strongly that this is still a duplication of test cases, I'm happy to remove it.

I meant to suggest that this previous case covering error conditions should be converted to cover the new error case(s) resulting from the changes rather than new positive cases.  I recognize that the current error condition of the case is invalidated by your changes but was hoping you would consider and identify the new ones.  I admit I didn't try to think about them and was attempting to duck the effort.

To be clear, the change is increasing the power of this capability and that's a good thing that I appreciate: thank you!  I appreciate the change but also your efforts.  I mostly mean, while trying not to do the work, to suggest some directions of consideration to support its success.  Perhaps for serving to your purposes to make suggestions that will increase the stickiness of the change (uncovered intentions have been removed in the past).

Reply:
This tests look fine to. PR relaxes so given error conditions are no longer error conditions, so tests were updated according to that.

With new proposal there seem to be no real error conditions (we accept any value as returned by module as valid), but I've proposed to apply some sanity restrictions here: https://github.com/serverless/serverless/pull/7540/files#r415727440